File: fixtures/scale/dart/large/lib/live/progress_reporter.dart
import '../core/suite_result.dart';
import 'console_sink.dart';

class ProgressReporter {
  final ConsoleSink sink = ConsoleSink();
  int _total = 0;
  int _done = 0;

  void begin(int suiteCount) {
    _total = suiteCount;
    sink.line('running $suiteCount suites');
  }

  void startSuite(String name) {
    sink.transientLine('[${_done + 1}/$_total] $name...');
  }

  void finishSuite(SuiteResult result) {
    _done++;
    final status = result.allPassed() ? 'ok' : 'FAIL';
    sink.line('\r[$_done/$_total] ${result.suiteName} $status');
  }

  void end(String summaryText) {
    sink.line(summaryText);
  }
}

File: fixtures/scale/kotlin/large/src/cryptobench/live/ProgressReporter.kt
package cryptobench.live

import cryptobench.core.SuiteResult

class ProgressReporter {
    private val sink = ConsoleSink(System.out)
    private var total = 0
    private var done = 0

    fun begin(suiteCount: Int) {
        total = suiteCount
        sink.line("running " + suiteCount + " suites")
    }

    fun startSuite(name: String) {
        sink.transientLine("[" + (done + 1) + "/" + total + "] " + name + "...")
    }

    fun finishSuite(result: SuiteResult) {
        done++
        val status = if (result.allPassed()) "ok" else "FAIL"
        sink.line("\r[" + done + "/" + total + "] " + result.suiteName + " " + status)
    }

    fun end(summaryText: String) {
        sink.line(summaryText)
    }
}
